### PR TITLE
Initial commit for final disposition updates

### PR DESCRIPTION
**Pull request by dborgen2**
## What’s changing?

Added the following checks client side and in the API
- Initial Action Taken cannot be null when Final Disposition is filled out
- Initial Action Date cannot be null when Final Disposition is filled out
- Final Disposition Date is required when Final Disposition is filled out
- Final Disposition is required when Final Disposition Date is filled out
- Final Disposition Date cannot be less than the match date

In addition, the following functionality was added:

- When Initial Action Taken isn't filled out, the Final Disposition area is disabled
- If Initial Action Taken is filled out, and Final Disposition is filled out, and then Initial Action Taken is cleared, Final Disposition will not be disabled. Instead an error will be thrown (see first two bullets above)
- Updated the PatchAsync response to not simply return a generic error. It will display the error returned from the API in a JSON failed response object. Then it's up to the client application to decide what to do with it. In our case, we add it to the server errors on top of the Match Detail page.
- If server-side errors do happen, we no longer throw away the values we tried submitting. We put them back in the form (but not on the right side)
- Default the Final Disposition Date when "Benefits Unchanged" or "Benefits Denied" was selected.

TL;DR
Final Disposition dropdown and date error validation, and disabling the section when Initial Action isn't yet filled in.

## Why?

Closes NAC-74 & NAC-592

## This PR has:

- [x] **Commented source code** (_required on public classes and methods, and elsewhere as appropriate_)

- [ ] **Developer documentation** (_for new build/test/API changes or complex portions of the system_)

- [x] **Automated unit tests** (_to maintain or increase level of code coverage_)

- [ ] **Changes to IAC** (_add any deployment steps that require manual intervention to the draft release notes_)

### CODE REVIEW

File: components/tests/Piipan.Components.Tests/Validation/UsaRequiredIfAttributeTests.cs [hunk 1]
```diff
@@ -4,54 +4,172 @@
...
+            Assert.Equal(hasError, !string.IsNullOrEmpty(validationResult?.ErrorMessage));
+        }
+
+        /// <summary>
+        /// Validate a UsaRequiredIf attribute attached to a dummy model using various parameters against an expected value
+        /// </summary>
+        [Theory]
+        [InlineData("expectedvalue", "test", false)]
+        [InlineData("expectedvalue", "", true)] // Error thrown when property 1 has expectedvalue and property 2 is empty
+        [InlineData("notexpectedvalue", "", false)] // Error NOT thrown when property 1 has an unexpected value and property 2 is empty
```
Comment (ccaprio97) [inline components/tests/Piipan.Components.Tests/Validation/UsaRequiredIfAttributeTests.cs:57]: does this equate to, we dont throw an error if theres bad data in intial action and no value in final disposition?

File: match/src/Piipan.Match/Piipan.Match.Core/Parsers/AddEventRequestParser.cs [hunk 2]
```diff
@@ -60,6 +60,10 @@ public async Task<AddEventRequest> Parse(Stream stream)
 
                 return request;
             }
+            catch (ValidationException)
+            {
+                throw;
```
Comment (ccaprio97) [inline match/src/Piipan.Match/Piipan.Match.Core/Parsers/AddEventRequestParser.cs:65]: do we not want to throw a certain type?

File: components/tests/Piipan.Components.Tests/Validation/UsaRequiredIfAttributeTests.cs (hunk 1 above)
Comment (dborgen2) [reply components/tests/Piipan.Components.Tests/Validation/UsaRequiredIfAttributeTests.cs:57]: No, good question. This right here is for verifying that if the source property has an expected value then we should be required. This actually isn't used in the app yet, but it is coming. This will be for the Invalid Match error stuff, and making sure that the Invalid Match For Other reason is required when Invalid Match Reason is "Other".

File: match/tests/Piipan.Match.Func.ResolutionApi.Tests/AddEventApiTests.cs [hunk 3]
```diff
@@ -115,13 +115,14 @@ public async void AddEvent_ReturnsUnauthorizedIfMatchClosed()
             var matchRecordDao = new Mock<IMatchRecordDao>();
             matchRecordDao
                 .Setup(r => r.GetRecordByMatchId(It.IsAny<string>()))
-                .ReturnsAsync(new MatchRecordDbo() { States = new string[] { "ea", "bc"}});
+                .ReturnsAsync(new MatchRecordDbo() { States = new string[] { "ea", "bc" } });
```
Comment (ccaprio97) [inline match/tests/Piipan.Match.Func.ResolutionApi.Tests/AddEventApiTests.cs:118]: dont know why spacing was added but not an issue

**Review by ccaprio97 — APPROVED** [review]
didnt see anything worthy of blocking so LGTM

File: components/tests/Piipan.Components.Tests/Validation/UsaRequiredIfAttributeTests.cs (hunk 1 above)
Comment (ccaprio97) [reply components/tests/Piipan.Components.Tests/Validation/UsaRequiredIfAttributeTests.cs:57]: gotcha

File: match/tests/Piipan.Match.Func.ResolutionApi.Tests/AddEventApiTests.cs (hunk 3 above)
Comment (dborgen2) [reply match/tests/Piipan.Match.Func.ResolutionApi.Tests/AddEventApiTests.cs:118]: Mm.. yea, probably the code cleanup that happens on save added it. Kind of annoying but should be fine
